tests/bam_io/test_bam_index.h

```diff
@@ -109,5 +124,79 @@ SEQAN_DEFINE_TEST(test_bam_io_bam_index_save)
     SEQAN_ASSERT(_compareBinaryFiles(toCString(tmpOutPath), toCString(baiFilename)));
 }
 
+SEQAN_DEFINE_TEST(test_bam_io_bam_index_view_records)
+{
+    CharString bamFileName(getAbsolutePath("/tests/bam_io/ex1.bam"));
+    CharString baiFileName(getAbsolutePath("/tests/bam_io/ex1.bam.bai"));
+
+    BamFileIn bamFile;  // Open BamFileIn for reading
+    SEQAN_ASSERT(open(bamFile, toCString(bamFileName)));
+
+    BamIndex<Bai> baiFile;  // Open BAI index file
+    open(baiFile, toCString(baiFileName));
+
+    BamHeader header;
+    readHeader(header, bamFile);
+
+    std::vector<BamAlignmentRecord> records;
+
+    viewRecords(records, bamFile, baiFile, 0, 1, 0);
```

Comment:
Nice test case :+1: . we should add this for jumpToRegion

Reply:
we should also add one with the region `[0, 100]` where it is clearly not 1-based.